client/etcd/client.go

```diff
@@ -240,8 +243,32 @@ func (c *csclient) etcdClientGen(zone string) (*clientv3.Client, error) {
 	return cli, nil
 }
 
-func newClient(endpoints []string) (*clientv3.Client, error) {
-	return clientv3.New(clientv3.Config{Endpoints: endpoints})
+func newClient(endpoints []string, certfile string, keyfile string, ca string) (*clientv3.Client, error) {
+	tlscfg := &tls.Config{}
+	if certfile != "" && keyfile != "" {
```

Comment:
should we return error when only one of certifile and keyfile is available?

Reply:
i'll just check certfile only for zero-value, then the error should be handled by the failure in tls.LoadX509KeyPair.